gpytorch/means/hadamard_multitask_mean.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+
+import torch
+from .multitask_mean import MultitaskMean
+
+
+class HadamardMultitaskMean(MultitaskMean):
```

Comment:
For `HadamardMultitaskMean` are you sure we want to be extending `MultitaskMean`? `MultitaskMean` returns `*batch_shape x n x t` tensors, assuming we have an observation for each data point for each task. Since in this case each data point along the `n` dimension belongs to a different task, I think we just want a `*batch_shape x n` result.

Maybe with the addition of `HadamardMultitaskMean` we should deprecate `MultitaskMean` and rename it `KroneckerMultitaskMean`.